server/app/models/user.rb

```diff
@@ -40,6 +40,7 @@ def get_advert_count
   end
 
   def jwt_payload
-    { "email" => self.email, "id" => self.id }
+    @role = Role.find(self.role_id)
+    { "email" => self.email, "id" => self.id, "role" => @role.name }
```

Comment:
it could be `self.role.name`, or `self.role_name` if you add delegation of name method to role association in the beginning of User model, e.g:

```
  delegate :name, to: :role, prefix: true
```
ref: https://api.rubyonrails.org/classes/Module.html#method-i-delegate
<img width="983" alt="Screenshot 2022-04-13 at 07 46 53" src="https://user-images.githubusercontent.com/201830/163102187-38b8c51d-9986-4de7-a29a-6de0e52fd461.png">

Reply:
As I can assume, we have not correct set_default method, so the association with role model not established. That's why delegate doesn't work to.
/user.rb
```
def set_default
    role_user = Role.find_by(name: "User")
    self.role_id = role_user.id
  end
```
So, in schema,rb
```
create_table "roles", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
    t.string "name"
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
  end
```
/user.rb
```
class User < ApplicationRecord
  has_one :role
end
```
/role.rb
```
class Role < ApplicationRecord
  has_many :users
  validates_uniqueness_of :name
end
```
if i change 
```
def set_default
    self.role = role_user
 end
```
this will return `ActiveModel::MissingAttributeError: can't write unknown attribute `user_id``
I don't know how to make this ok but in documentation:
![зображення](https://user-images.githubusercontent.com/82406343/163344885-3ffe9643-0d2a-4dcc-8bd3-58d2f08dd77f.png)
![зображення](https://user-images.githubusercontent.com/82406343/163344933-90889944-79be-4769-8d02-e84f36e352c9.png)
So, i actually don't know how to make relations correctly, because has_one in user model seems not to be good. Theare methods were written earlier as association too